File: 6.c
#include <stdio.h>
int main()
{
	int N, p, q, i, dem, j;
	printf("Nhap so N duong ");
	scanf("%d", &dem);

	for (j = 2; j < dem; j++)
	{
		p = q = 0;
		N = j;

		for (i = 1; i < N; i++)
		{
			if (N % i == 0)
			{
				p = p + i;
			}
		}

		for (i = 1; i < p; i++)
		{
			if (p % i == 0)
			{
				q = q + i;
			}
		}

		if (q == N && p != q)
			printf("Cap so do la: %d  %d\n", N, p);
	}
}

File: python/6.py
temp = int(input("nhap n:"))
p = q = 0
for i in range(2, temp):
    p = q = 0
    n = i
    for j in range(1, n):
        if n % j == 0:
            p += j
    for h in range(1, p):
        if p % h == 0:
            q += h

    if q == n and p != q:
        print("cap so do la: ", n, p)
